fendermint/app/src/settings/resolver.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::{path::PathBuf, time::Duration};
+
+use serde::Deserialize;
+use serde_with::{serde_as, DurationSeconds};
+
+use ipc_sdk::subnet_id::SubnetID;
+use multiaddr::Multiaddr;
+
+use super::IsHumanReadable;
+use crate::home_relative;
+
+#[serde_as]
+#[derive(Debug, Clone, Deserialize)]
+pub struct ResolverSettings {
+    /// Time to wait between attempts to resolve a CID, in seconds.
+    #[serde_as(as = "DurationSeconds<u64>")]
+    pub retry_delay: Duration,
+    /// The current subnet ID for which this node can serve data.
+    ///
+    /// TODO #231: Take this from the IPC settings.
+    #[serde_as(as = "IsHumanReadable")]
+    pub subnet_id: SubnetID,
+
+    pub network: NetworkSettings,
+    pub discovery: DiscoverySettings,
+    pub membership: MembershipSettings,
+    pub connection: ConnectionSettings,
+    pub content: ContentSettings,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct NetworkSettings {
+    /// Cryptographic key used to sign messages.
+    ///
+    /// This is the name of a Secp256k1 private key file,
+    /// relative to the `home_dir`.
+    local_key: PathBuf,
```

Comment:
This is the private key used for the libp2p host identity, i.e. the one that determines the PeerID?

Reply:
That's right, see: 
* https://github.com/consensus-shipyard/ipc-ipld-resolver/blob/main/src/behaviour/mod.rs#L30-L35
* https://github.com/consensus-shipyard/ipc-ipld-resolver/blob/main/src/service.rs#L150-L151
* https://github.com/consensus-shipyard/ipc-ipld-resolver/blob/main/src/service.rs#L169